neuland/digitizing/NeulandPointFilter.cxx

```diff
@@ -2,45 +2,68 @@
 #include <iostream>
 #include <utility>
 
-constexpr auto proton_pid = 2212;
+using ParticleUType = std::underlying_type<BitsetParticle>::type;
 
-void NeulandPointFilter::add_filtered_pid(int filtered_pid, std::unordered_map<int, int>& filtered_pids_hash)
+constexpr auto ParticleToBitset(BitsetParticle particle)
 {
-    filtered_pids_hash.emplace(filtered_pid, filtered_pid);
+    return std::bitset<ParticleBitsetSize>{ static_cast<ParticleUType>(particle) };
 }
-void NeulandPointFilter::clear_filtered_pids_hash(std::unordered_map<int, int>& filtered_pids_hash)
+
+auto BitsetToParticle(std::bitset<ParticleBitsetSize> bits) -> BitsetParticle
 {
-    filtered_pids_hash.clear();
+    return static_cast<BitsetParticle>(static_cast<uint32_t>(bits.to_ulong()));
 }
+
+auto CheckCriteria(BitsetParticle particle, BitsetParticle criteria) -> bool
+{
+    return (ParticleToBitset(particle) & ParticleToBitset(criteria)) == ParticleToBitset(particle);
+}
+
+auto operator|(BitsetParticle left, BitsetParticle right) -> BitsetParticle
+{
+    auto left_bitset = ParticleToBitset(left);
+    auto right_bitset = ParticleToBitset(right);
+    return BitsetToParticle(left_bitset | right_bitset);
+}
+
+auto operator&(BitsetParticle left, BitsetParticle right) -> BitsetParticle
+{
+    auto left_bitset = ParticleToBitset(left);
+    auto right_bitset = ParticleToBitset(right);
+    return BitsetToParticle(left_bitset & right_bitset);
+}
+
+auto operator~(BitsetParticle particle) -> BitsetParticle { return BitsetToParticle(~ParticleToBitset(particle)); }
+
+constexpr auto proton_pid = 2212;
+constexpr auto neutron_pid = 2112;
+constexpr auto electron_pid = 11;
+constexpr auto positron_pid = -11;
+constexpr auto alpha_pid = 1000040020;
+constexpr auto gamma_pid = 22;
+
+auto NeulandPointFilter::check_pid_for_criteria(int pid, BitsetParticle criteria) -> bool
+{
+    return ((CheckCriteria(BitsetParticle::proton, criteria) && pid == proton_pid) ||
+            (CheckCriteria(BitsetParticle::neutron, criteria) && pid == neutron_pid) ||
+            (CheckCriteria(BitsetParticle::electron, criteria) && pid == electron_pid) ||
+            (CheckCriteria(BitsetParticle::positron, criteria) && pid == positron_pid) ||
+            (CheckCriteria(BitsetParticle::alpha, criteria) && pid == alpha_pid) ||
+            (CheckCriteria(BitsetParticle::gamma, criteria) && pid == gamma_pid));
+}
```

Comment:
You should write another conversion function that converts the PID to BitsetParticle.

I could think of two ways:
1. Use std::unordered_map
```cpp
auto PID_to_BitSetParticle = std::unordered_map<int, BitsetParticle>{{2212, BitsetParticle::proton}, {2112, BitsetParticle::neutron}};
```
And then in the conversion function, you get the particle from its PID with
```cpp
auto particle = PID_to_BitSetParticle.at(pid);
```
2. Use switch case:
```cpp
switch(pid){
case 2212:
    return BitsetParticle::proton; break;
case 2112:
    return BitsetParticle::neutron;break;
}
```